### PR TITLE
Add TimeGrad diffusion model

### PR DESCRIPTION
**Pull request by alyssadsouza**
## Summary
- Adds TimeGrad as a `BaseTimeSeriesFoundationModel` child class for holdout evaluation
  - Trains from scratch on CGM data as a non-foundation-model baseline for comparison
- Adds `--skip-steps` flag to the generic holdout workflow for per-step control and auto-skips zero-shot for models that train from-scratch (no zero-shot implementation exists which is the case with TimeGrad)

## Changes
- **`src/models/timegrad/`**: new model package (config, model, init) wrapping the `pytorchts` TimeGradEstimator
- **`scripts/examples/example_holdout_generic_workflow.py`**: register timegrad in ModelFactory, replace `--skip-training` branching with flat `--skip-steps` checks
- **`scripts/setup_model_env.sh`**: support per-model Python version (TimeGrad requires 3.11)
- **`pyproject.toml`**: add `timegrad` optional dependency group
- **`docs-internal/notebooks/4.15-ad-timegrad.ipynb`**: exploration notebook

## Notes
- Requires Python 3.11 (`pytorchts` is abandoned, pinned to old gluonts/pandas without 3.12 wheels)
- Monkey-patches pytorchts for PyTorch 2.x compatibility (circular padding + CondUpsampler for univariate case)

## Testing
- [ ] Run `setup_model_env.sh timegrad` to create venv
- [ ] Run holdout workflow: `MODEL_TYPE="timegrad" SKIP_TRAINING="false" EPOCHS=1   ./scripts/examples/run_holdout_generic_workflow.sh`
- [ ] Verify steps 5 (train), 6 (load checkpoint + predict), 7 (resume training) succeed
- [ ] Verify step 4 auto-skips with log message
- [ ] Verify `--skip-steps` works for other model types (e.g., `--skip-steps 7` with TTM)

### CODE REVIEW

**Comment by alyssadsouza** [comment]
@RobotPsychologist regarding using the HF API here I don't think it would make sense because TimeGrad trains from scratch, so there are no pretrained weights on HF to load from.

**Review by RobotPsychologist — APPROVED** [review]
@alyssadsouza Great work!

File: scripts/experiments/sliding_window_eval.py [hunk 1]
```diff
@@ -14,6 +14,15 @@
     python scripts/experiments/sliding_window_eval.py --model sundial --context-length 512 --forecast-length 72
     python scripts/experiments/sliding_window_eval.py --model sundial --model-config configs/models/sundial.yaml
     python scripts/experiments/sliding_window_eval.py --model sundial --checkpoint path/to/checkpoint
+
+    # TimeGrad — after first 10-epoch training run (lynch_2022):
+d
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply scripts/experiments/sliding_window_eval.py]: Fixed in next push. This was a copy-paste error.

File: scripts/examples/run_holdout_generic_workflow.sh [hunk 2]
```diff
@@ -33,6 +33,9 @@
 #
 # All datasets combined with specific model:
 #   CUDA_VISIBLE_DEVICES=1 MODEL_TYPE="ttm" MODEL_CONFIG="configs/models/ttm/fine_tune.yaml" CONFIG_DIR="configs/data/holdout_10pct" DATASETS="lynch_2022 aleppo_2017 brown_2019" SKIP_TRAINING="false" ./scripts/examples/run_holdout_generic_workflow.sh
+#
+# TimeGrad, all datasets, CGM only (10 epochs):
+#   MODEL_TYPE="timegrad" MODEL_CONFIG="configs/models/timegrad/cgm_only.yaml" CONFIG_DIR="configs/data/holdout_10pct" EPOCHS=10 DATASETS="lynch_2022 aleppo brown_2019 tamborlane_2008" SKIP_TRAINING="false" ./scripts/examples/run_holdout_generic_workflow.sh
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply scripts/examples/run_holdout_generic_workflow.sh]: Fixed in next push.

File: scripts/examples/example_holdout_generic_workflow.py [hunk 3]
```diff
@@ -1710,13 +1773,26 @@ def main():
...
+    # Build the set of steps to skip
+    skip_steps: set[int] = set(args.skip_steps)
+    if args.skip_training:
+        skip_steps.update({5, 6, 7})
+    # Auto-skip zero-shot for models that train from scratch (no pretrained weights)
+    if not ModelFactory.get_default_model_path(args.model_type):
+        if 4 not in skip_steps:
+            logger.info(
+                f"Auto-skipping step 4 (zero-shot): {args.model_type} trains from "
+                f"scratch and has no pretrained weights"
```
[the remark this answers is not in the document]
Comment (RobotPsychologist) [reply scripts/examples/example_holdout_generic_workflow.py]: @alyssadsouza this is the last thing we should check, then we're good to merge.
